cpp/src/experimental/louvain.cuh

```diff
@@ -405,67 +60,35 @@ class Louvain {
       handle_(handle),
       dendrogram_(std::make_unique<Dendrogram<vertex_t>>()),
       current_graph_view_(graph_view),
-      compute_partition_(handle, graph_view),
-      local_num_vertices_(graph_view.get_number_of_local_vertices()),
-      local_num_rows_(graph_view.get_number_of_local_adj_matrix_partition_rows()),
-      local_num_cols_(graph_view.get_number_of_local_adj_matrix_partition_cols()),
-      local_num_edges_(graph_view.get_number_of_edges()),
-      vertex_weights_v_(graph_view.get_number_of_local_vertices()),
-      cluster_weights_v_(graph_view.get_number_of_local_vertices()),
-      number_of_vertices_(graph_view.get_number_of_local_vertices()),
+      cluster_keys_v_(graph_view.get_number_of_local_vertices(), handle.get_stream()),
+      cluster_weights_v_(graph_view.get_number_of_local_vertices(), handle.get_stream()),
+      vertex_weights_v_(graph_view.get_number_of_local_vertices(), handle.get_stream()),
+      src_vertex_weights_cache_v_(0, handle.get_stream()),
+      src_cluster_cache_v_(0, handle.get_stream()),
+      dst_cluster_cache_v_(0, handle.get_stream()),
       stream_(handle.get_stream())
   {
     if (graph_view_t::is_multi_gpu) {
-      rank_               = handle.get_comms().get_rank();
-      base_vertex_id_     = graph_view.get_local_vertex_first();
-      base_src_vertex_id_ = graph_view.get_local_adj_matrix_partition_row_first(0);
-      base_dst_vertex_id_ = graph_view.get_local_adj_matrix_partition_col_first(0);
-
-      local_num_edges_ = thrust::transform_reduce(
-        thrust::host,
-        thrust::make_counting_iterator<size_t>(0),
-        thrust::make_counting_iterator<size_t>(
-          graph_view.get_number_of_local_adj_matrix_partitions()),
-        [&graph_view](auto indx) {
-          return graph_view.get_number_of_local_adj_matrix_partition_edges(indx);
-        },
-        size_t{0},
-        thrust::plus<size_t>());
-
-      CUDA_TRY(cudaStreamSynchronize(stream_));
-    }
-
-    src_indices_v_.resize(local_num_edges_);
-
-    cugraph::detail::offsets_to_indices(
-      current_graph_view_.offsets(), local_num_rows_, src_indices_v_.data().get());
-
-    if (base_src_vertex_id_ > 0) {
-      thrust::transform(rmm::exec_policy(stream_)->on(stream_),
-                        src_indices_v_.begin(),
-                        src_indices_v_.end(),
-                        thrust::make_constant_iterator(base_src_vertex_id_),
-                        src_indices_v_.begin(),
-                        thrust::plus<vertex_t>());
+      rank_           = handle.get_comms().get_rank();
```

Comment:
What's the purpose of creating a separate copy of rank_ here? This will add burden to remember that ```rank_``` and ```handle.get_comms().get_rank()``` are same; maintain those two variables in sync; and this will eat-up additional register/cache line/memory.

Reply:
Removed